compiler/optimizer/OMRValuePropagation.hpp

```diff
@@ -281,118 +284,32 @@ class ValuePropagation : public TR::Optimization
    ValueConstraint *copyValueConstraints(ValueConstraints &valueConstraints);
 
 #if USE_TREES
-   class ValueConstraintHandler : public TREE_HANDLER<ValueConstraint>
+   class ValueConstraintHandler : public TREE_HANDLER <ValueConstraint>
       {
       public:
-      void setVP(TR::ValuePropagation *vp) {_vp = vp;}
```

Comment:
No, so we can use OMR::ValuePropagation in OMR layer, and J9::ValuePropagation in J9 layer, which means we do not need the namespace mapping, each layer will be talking about their version of VP since it's virtual inheritance. The only thing is that GVP and LVP need to subclass from the right VP, so we still need something similar to mapping, we're thinking about using a define, for example in ValuePropagation.hpp in trj9
#define TR::ValuePropagation J9::ValuePropagation

What do you think? Is there a known pattern to resolve this issue?